Source/WebKit/Shared/unix/BreakpadExceptionHandler.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * If not stated otherwise in this file or this component's license file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 RDK Management
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if ENABLE(BREAKPAD)
+#include "config.h"
+#include "BreakpadExceptionHandler.h"
+
+#include <client/linux/handler/exception_handler.h>
+#include <mutex>
+#include <signal.h>
+#include <wtf/FileSystem.h>
+#include <wtf/NeverDestroyed.h>
+
+
+namespace WebKit {
+
+static std::once_flag onceFlag;
+
+void installBreakpadExceptionHandler()
+{
+    static WTF::MainThreadLazyNeverDestroyed<google_breakpad::ExceptionHandler> exceptionHandler;
+    static String breakpadMinidumpDir = String::fromUTF8(getenv("BREAKPAD_MINIDUMP_DIR"));
+
+#ifdef BREAKPAD_MINIDUMP_DIR
+    if (breakpadMinidumpDir.isEmpty())
+        breakpadMinidumpDir = StringImpl::createFromCString(BREAKPAD_MINIDUMP_DIR);
+#endif
+
+    if (breakpadMinidumpDir.isEmpty())
+        return;
+
+    if (FileSystem::fileType(breakpadMinidumpDir) != FileSystem::FileType::Directory) {
+        fprintf(stderr, "Breakpad dir \"%s\" is not a directory, not installing handler\n", breakpadMinidumpDir.utf8().data());
```

Comment:
Replaced the `WTFLogAlways` from previous iteration as when this code is run WebKit is not yet initialized.

Reply:
Oops. Didn't know that. Thanks.